graphql/util.go

```diff
@@ -427,46 +427,37 @@ func mapHTTPStatusToGqlError(ctx context.Context, httpStatus int, err error) *gq
 	}
 }
 
-func isTaskBlocked(ctx context.Context, at *restModel.APITask) (*bool, error) {
-	t, err := task.FindOneIdNewOrOld(*at.Id)
-	if err != nil {
-		return nil, ResourceNotFound.Send(ctx, err.Error())
+func canRestartTask(t *task.Task) bool {
+	// Cannot restart execution tasks.
+	isExecTask := t.IsPartOfDisplay()
+	if isExecTask {
+		return false
 	}
-	if t == nil {
-		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("task %s not found", *at.Id))
+	if !utility.StringSliceContains(evergreen.TaskUncompletedStatuses, t.Status) {
+		return true
 	}
-	isBlocked := t.Blocked()
-	return &isBlocked, nil
-}
-
-func isExecutionTask(ctx context.Context, at *restModel.APITask) (*bool, error) {
-	i, err := at.ToService()
-	if err != nil {
-		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error while converting task %s to service", *at.Id))
+	if t.Aborted {
+		return true
 	}
-	t, ok := i.(*task.Task)
-	if !ok {
-		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Unable to convert APITask %s to Task", *at.Id))
+	if t.DisplayStatus == evergreen.TaskStatusBlocked && t.DisplayOnly {
+		return true
 	}
-	isExecutionTask := t.IsPartOfDisplay()
-
-	return &isExecutionTask, nil
+	return false
 }
 
-func canRestartTask(ctx context.Context, at *restModel.APITask) (*bool, error) {
-	taskBlocked, err := isTaskBlocked(ctx, at)
-	if err != nil {
-		return nil, err
+func canScheduleTask(t *task.Task) bool {
+	// Cannot schedule execution tasks.
+	isExecTask := t.IsPartOfDisplay()
```

Comment:
same comments here  -- or in this case, `if t.IsPartOfDisplay() || t.Aborted` is pretty readable.

There's no rule around this but I think generally boolean logic is readable if it's not more complicated than a "this and/or this"